elastic/kibana
# [POC] SO Metadata events streams

This PR is a POC for the content management "metadata events".

## Goal

We want to add a **new sorting capability** to the saved object list view tables to help our users quickly find the content they are looking for. We want to be able to sort by “**Total views**” (ASC and DESC) and also “Total views in the last 30 / 15 / 7 days” (ASC and DESC).

In this PR I demo how we could add views counters (for last "7_days", "14_days", ...) and be able to sort the Dashboard list view table with those counters.

## How to test

* Add the sample data (this will add 3 dashboards)
* Navigate to the dashboard list view. There should be a column with the view count + a drop down in the table header to change the granularity of the counts
* Click a dashboard to view it and go back to the dashboard list. Its counter should now be `1`

In order to test the granularity we need to manually add an event in the past. In a terminal send the following curl command (**note:** set the correct url of your local dev Kibana)

```
curl -X "POST" "http://localhost:5601/<unique>/api/user_content/event/viewed:kibana" \
     -H 'kbn-xsrf: kibana' \
     -H 'Content-Type: application/json; charset=utf-8' \
     -u 'elastic:changeme' \
     -d $'{
  "@timestamp": "2022-05-15T10:00:00.000Z", # change this date if needed
  "soId": "1bcfb990-d859-11ec-8a6b-b125ca46a567", # change if needed
  "soType": "dashboard"
}'
``` 

You should now have an event in the past and be able to test the other granularity ("30 days", "45 days"...)

Issue related: https://github.com/elastic/kibana/issues/9202
